crates/core_arch/src/aarch64/neon/generated.rs

```diff
@@ -1601,6 +1601,42 @@ pub unsafe fn vcvtpq_u64_f64(a: float64x2_t) -> uint64x2_t {
     vcvtpq_u64_f64_(a)
 }
 
+/// Negate
+#[inline]
+#[target_feature(enable = "neon")]
+#[cfg_attr(test, assert_instr(sub))]
+pub unsafe fn vneg_s64(a: int64x1_t) -> int64x1_t {
+    let b: i64x1 = i64x1::new(0);
+    simd_sub(transmute(b), a)
```

Comment:
Nice, I was actually looking into this one as well earlier and "gave up on it", because there didn't seem to be a way other than 0  - the_vec. Good to see I didn't miss anything xD

Reply:
It seems that core_arch does not have an instruction such as simd_neg, I don’t know how to add it, because I don’t know the compiler very well. :(

Reply:
You will need to add to the function at https://github.com/rust-lang/rust/blob/f24ce9b0140d9be5a336954e878d0c1522966bb8/compiler/rustc_codegen_llvm/src/intrinsic.rs#L730 It is probably similar to https://github.com/rust-lang/rust/blob/f24ce9b0140d9be5a336954e878d0c1522966bb8/compiler/rustc_codegen_llvm/src/intrinsic.rs#L1631-L1661 except a single argument instead of two arguments.

Reply:
Thanks for guiding. I will try later. I think the implementation of sub(0, a) is sufficient for now. Since the implementation in Clang is the [same](https://godbolt.org/z/e956hK)

Reply:
They are different if you look at the code for floating-point numbers. `0.0 - x` and `-x` have different behavior for floats with regards to negative zeros, NaNs, etc.

Reply:
It looks like we do need to add simd_neg. I suggest to close this PR, and then raise another issue to track its implementation.